### PR TITLE
Support Faraday v2

### LINKED ISSUE #488: Support Faraday 2.x
[Faraday 2.x has been released.](https://github.com/lostisland/faraday/blob/fc112254746ae92e3dcc8e6f9055878b1c583cc7/CHANGELOG.md#200-2022-01-04)

but, faraday is require to `< 2.0.0` in the gemspec of zendesk_api.
https://github.com/zendesk/zendesk_api_client_rb/blob/8665b8e9bfb638805be550ab54aca0ac59b86271/zendesk_api.gemspec#L29

Is it possible to relax the dependency?

### PR DESCRIPTION
**Pull request by ecoologic**
A very popular request for our Gem. Support for Faraday 2, which has been released early in the year. We are preventing our users from updating, but, no more! This BETA release allows users to update Faraday.

We expect to leave this feature in BETA for 6 months during which we can address any problem our customers have with the update. The release will follow after the PR is merged.

Please let us know your thoughts and any problems you encounter.

Fixes #488.

---

**Note**: At the moment JRuby 9.1 and 9.2 are not supported in this version. Further investigation _might_ find a fix, but it's likely we will stop testing against JRuby 9.1 and JRuby 9.2.

See [Faraday Documentation](https://www.rubydoc.info/gems/faraday_middleware/1.2.0#as-of-v0-16-0-faraday-and-faraday_middleware-no-longer-officially-support-jruby-or-rubinius):

> As of v0.16.0, faraday and faraday_middleware no longer officially support JRuby or Rubinius.

Our tests kept working until this upgrade, but unfortunately, both [JRuby 9.1](https://github.com/zendesk/zendesk_api_client_rb/runs/8110095881) and [JRuby 9.2](https://github.com/zendesk/zendesk_api_client_rb/runs/8110151024) stopped working.

However, it is unclear if the latest v1 actually does support [JRuby 9.1](https://github.com/zendesk/zendesk_api_client_rb/runs/8127916517?check_suite_focus=true#step:4:117) or [JRuby 9.2](https://github.com/zendesk/zendesk_api_client_rb/runs/8127998606?check_suite_focus=true#step:4:117).

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -18,8 +18,7 @@ jobs:
           - 2.7
           - 3.0
           - 3.1
-          - jruby-9.1
-          - jruby-9.2
```
Comment (ecoologic) [inline .github/workflows/main.yml:22]: These are already not _really_ working in master, see description for details.

File: lib/zendesk_api/verbs.rb [hunk 2]
```diff
@@ -44,14 +44,12 @@ def create_verb(method_verb)
...
-              false
-            rescue ZendeskAPI::Error::ClientError
-              false
-            end
+            send("#{method}!", *method_args)
+          rescue ZendeskAPI::Error::RecordInvalid => e
+            @errors = e.errors
+            false
+          rescue ZendeskAPI::Error::ClientError
+            false
```
Comment (ecoologic) [inline lib/zendesk_api/verbs.rb:52]: I'm not really happy that someone looking at these lines might think I wrote this, and now just reformatted for 2.6 syntax. 🤷

File: spec/fixtures/zendesk.rb [hunk 3]
```diff
@@ -90,7 +90,9 @@ def group
 
     def organization
       VCR.use_cassette('valid_organization') do
-        @organization ||= current_user.organization
+        current_user.organization
+        # ... or, if you can't work it out locally:
+        # @organization ||= current_user.organization || client.organizations.fetch!.last
```
Comment (ecoologic) [inline spec/fixtures/zendesk.rb:95]: There's quite some work to do when it comes to improving DX, but I'm running out of time and for now this is the best I can do.

File: lib/zendesk_api/middleware/response/gzip.rb [hunk 4]
```diff
@@ -7,11 +7,12 @@ module Middleware
...
     module Response
       # Faraday middleware to handle content-encoding = gzip
-      class Gzip < Faraday::Response::Middleware
+      class Gzip < Faraday::Middleware
         def on_complete(env)
-          if !env[:body].strip.empty? && env[:response_headers]['content-encoding'] == "gzip"
-            env[:body] = Zlib::GzipReader.new(StringIO.new(env[:body])).read
-          end
+          return if env[:response_headers]['content-encoding'] != "gzip"
+          return if env[:body].force_encoding(Encoding::BINARY).strip.empty?
```
Comment (ecoologic) [inline lib/zendesk_api/middleware/response/gzip.rb:13]: Encoding is required for `strip`. I can't think of a valid use case why we would need that, but, who's got the guts to remove it and find out they're wrong?

**Review by Kalimaha — APPROVED** [review]
LGTM 👍

**Review by codealchemy — APPROVED** [review]

**Comment by rsl** [comment]
@ecoologic thanks for yr fast turn around on this! we're putting it out in production to help test [as i type]. thanks again so much.

**Pull request merged**